examples/openshift-origin/create.sh

```diff
@@ -14,18 +14,103 @@
+	exit 1
+fi
+
+if [ ! $(which gcloud) ]
+then
+	echo "gcloud is a prerequisite for running this script. Please install gcloud and try running again."
+	exit 1
+fi
+
+gcloud_instances=$(gcloud compute instances list | grep "\-master")
+if [ -z "$gcloud_instances" ] || [ -z "${KUBE_GCE_INSTANCE_PREFIX}" ]
+then
+	echo "This script is only able to supply the necessary serviceaccount key if you are running on Google"
+	echo "Comput Engine using a cluster/kube-up.sh script with KUBE_GCE_INSTANCE_PREFIX set. If this is not"
+	echo "the case, be ready to supply a path to the serviceaccount public key."
+	if [ -z "${KUBE_GCE_INSTANCE_PREFIX}" ]
+	then
+		echo "Please provide your KUBE_GCE_INSTANCE_PREFIX now:"
+		read KUBE_GCE_INSTANCE_PREFIX
+	fi
+fi
+
 export OPENSHIFT_EXAMPLE=$(pwd)/examples/openshift-origin
+echo Set OPENSHIFT_EXAMPLE=${OPENSHIFT_EXAMPLE}
 export OPENSHIFT_CONFIG=${OPENSHIFT_EXAMPLE}/config
+echo Set OPENSHIFT_CONFIG=${OPENSHIFT_CONFIG}
 mkdir ${OPENSHIFT_CONFIG}
-cluster/kubectl.sh config view --output=yaml --flatten=true --minify=true > ${OPENSHIFT_CONFIG}/kubeconfig
-cluster/kubectl.sh create -f $OPENSHIFT_EXAMPLE/openshift-service.yaml
-sleep 60
-export PUBLIC_IP=$(cluster/kubectl.sh get services openshift --template="{{ index .status.loadBalancer.ingress 0 \"ip\" }}")
-echo "PUBLIC IP: ${PUBLIC_IP}"
-docker run --privileged -v ${OPENSHIFT_CONFIG}:/config openshift/origin start master --write-config=/config --kubeconfig=/config/kubeconfig --master=https://localhost:8443 --public-master=https://${PUBLIC_IP}:8443
-sudo -E chown ${USER} -R ${OPENSHIFT_CONFIG}
-docker run -i -t --privileged -e="OPENSHIFTCONFIG=/config/admin.kubeconfig" -v ${OPENSHIFT_CONFIG}:/config openshift/origin ex bundle-secret openshift-config -f /config &> ${OPENSHIFT_EXAMPLE}/secret.json
-cluster/kubectl.sh create -f ${OPENSHIFT_EXAMPLE}/secret.json
-cluster/kubectl.sh create -f ${OPENSHIFT_EXAMPLE}/openshift-controller.yaml
-cluster/kubectl.sh get pods | grep openshift
+echo Made dir ${OPENSHIFT_CONFIG}
+echo
+
+echo "===> Setting up etcd-discovery:"
+# A token etcd uses to generate unique cluster ID and member ID. Conforms to [a-z0-9]{40}
+export ETCD_INITIAL_CLUSTER_TOKEN=$(python -c "import string; import random; print(''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits) for _ in range(40)))")
+
+# A unique token used by the discovery service. Conforms to etcd-cluster-[a-z0-9]{5}
+export ETCD_DISCOVERY_TOKEN=$(python -c "import string; import random; print(\"etcd-cluster-\" + ''.join(random.SystemRandom().choice(string.ascii_lowercase + string.digits) for _ in range(5)))")
+sed -i.bak -e "s/INSERT_ETCD_INITIAL_CLUSTER_TOKEN/\"${ETCD_INITIAL_CLUSTER_TOKEN}\"/g" -e "s/INSERT_ETCD_DISCOVERY_TOKEN/\"${ETCD_DISCOVERY_TOKEN}\"/g" ${OPENSHIFT_EXAMPLE}/etcd-controller.yaml
+
+kubectl create -f ${OPENSHIFT_EXAMPLE}/etcd-discovery-controller.yaml
```

Comment:
Create all this stuff in a new namespace for the example.

Maybe call it "openshift-origin"

In this way, clean-up is just removal of the namespace.